### PR TITLE
feat(crc-jca): enable code reusability between family matters applications

### PR DESCRIPTION
**Pull request by VilliGunn**
# \<Description\>

We are building two very similar applications for the District Magistrate of Iceland regarding family matters. Since the flows are so similar we want to be able to reuse some of the code.

## What

- Added a new `family-matters-core` library that can be used for shared components utils etc.
- Enabled reuse of types, utils and component

TODO: Share data providers, custom fields and more.

## Why

- Be able to reuse code between similar applications

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/family-matters/joint-custody-agreement/types/index.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export type JCAFieldBaseProps = Override<
+  FieldBaseProps,
+  { application: JCAApplication; errors: ErrorSchema }
+>
+
+export enum DataProviderTypes {
+  MockNationalRegistry = 'MockNationalRegistryProvider',
+  NationalRegistry = 'NationalRegistryProvider',
+  UserProfile = 'UserProfileProvider',
+}
```
Comment (Rikkiabb) [inline libs/application/templates/family-matters/joint-custody-agreement/types/index.ts:31]: Talked about this in IRL but maybe this type will be moved to shared later and the providers as well.

Comment (VilliGunn) [reply libs/application/templates/family-matters/joint-custody-agreement/types/index.ts:31]: Yes, just want this PR merged to get code ownership and then we can move things around

File: SUMMARY.md [hunk 2]
```diff
@@ -124,9 +124,10 @@
   - [Data Providers](libs/application/data-providers/README.md)
   - [Template API Modules](libs/application/template-api-modules/README.md)
   - [Template Loader](libs/application/template-loader/README.md)
-    - [Children Residence Change Template](libs/application/templates/children-residence-change/README.md)
     - [Application Templates Data Protection Complaint](libs/application/templates/data-protection-complaint/README.md)
     - [Document Provider Onboarding](libs/application/templates/document-provider-onboarding/README.md)
+      - [Children Residence Change Template](libs/application/templates/family-matters/children-residence-change/README.md)
```
Comment (jeremybarbet) [inline SUMMARY.md:129]: Missing a README at the root of family-matters https://docs.devland.is/repository/gitbook-contribute#root-readme-md-within-handbook-directory

**Review by jeremybarbet — APPROVED** [review]
LGTM

**Pull request merged**